presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -1242,4 +1307,76 @@ private static boolean envelopes(Slice left, Slice right, EnvelopesPredicate pre
     {
         boolean apply(Envelope left, Envelope right);
     }
+
+    private static Iterable<Slice> getGeometrySlicesFromBlock(Block block)
+    {
+        requireNonNull(block, "block is null");
+        return () -> new Iterator<Slice>() {
+            private final int blockPositionCount = block.getPositionCount();
+            private int iteratorPosition;
+            @Override
+            public boolean hasNext()
+            {
+                return iteratorPosition != blockPositionCount;
```

Comment:
Why not call `block.getPositionCount()` here instead of having a `final` variable? Does it make any diff in perf?